src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c

```diff
@@ -338,12 +339,9 @@ void ADIOI_LUSTRE_WriteStridedColl(ADIO_File fd, const void *buf, int count,
 
     /* free all memory allocated for collective I/O */
     /* free others_req */
-    ADIOI_Free(others_req[0].offsets);
-    ADIOI_Free(others_req[0].mem_ptrs);
-    ADIOI_Free(others_req);
-    ADIOI_Free(buf_idx[0]);     /* also my_req[*].offsets and my_req[*].lens */
```

Comment:
Shouldn't we free what's in buf_idx[0]?
I believe it's necessary to free it in order to take of this allocation:

ADIOI_LUSTRE_Calc_my_req(ADIO_File fd, ADIO_Offset * offset_list,
...
196     ptr = (ADIO_Offset *) ADIOI_Malloc((memLen * 3 + nprocs) * sizeof(ADIO_Offset));
...
200     buf_idx[0] = ptr;